File: packages/grafana-faro/android/src/main/java/io/capawesome/capacitorjs/plugins/grafanafaro/classes/options/ViewMetadata.java
package io.capawesome.capacitorjs.plugins.grafanafaro.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.grafanafaro.GrafanaFaroPlugin;

public class ViewMetadata {

    @NonNull
    private final String name;

    public ViewMetadata(@NonNull PluginCall call) throws Exception {
        String name = call.getString("name");
        if (name == null || name.isEmpty()) {
            throw new Exception(GrafanaFaroPlugin.ERROR_NAME_MISSING);
        }
        this.name = name;
    }

    public ViewMetadata(@NonNull JSObject source) throws Exception {
        String name = source.getString("name");
        if (name == null || name.isEmpty()) {
            throw new Exception(GrafanaFaroPlugin.ERROR_NAME_MISSING);
        }
        this.name = name;
    }

    @NonNull
    public String getName() {
        return name;
    }
}

File: packages/grafana-faro/ios/Plugin/Classes/Options/ViewMetadata.swift
import Capacitor
import Foundation

@objc public class ViewMetadata: NSObject {
    private let name: String

    init(call: CAPPluginCall) throws {
        guard let name = call.getString("name"), !name.isEmpty else {
            throw CustomError.nameMissing
        }
        self.name = name
    }

    init(source: JSObject) throws {
        guard let name = source["name"] as? String, !name.isEmpty else {
            throw CustomError.nameMissing
        }
        self.name = name
    }

    func getName() -> String {
        return name
    }
}
